moderntribe/square1-global-docker
# Release 5.1.0

### Changelog

- [UPDATED]: global docker-compose.yml to add a fixed version for the Nginx proxy image (0.9.0).
- [ADDED]: Docker health checking to the MariaDB container (`docker ps` will show the health status of the container)
- [ADDED]: `so bootstrap --multisite` to install WP core as multisite
- [FIXED]: Building frontend assets with `so bootstrap` or `so create` due to yarn missing for specific node versions.
- [FIXED]: Wait for health check to complete for the MariaDB container before trying to create new databases during `so bootstrap/so create`
- [FIXED]: `--only-new` error showing up when running a command that doesn't exist.
- [ADDED]: CLI spinner for longer running processes like building the frontend

### Testing Instructions

- Check out this branch, `composer install` and verify `sodev` or `./so` shows version `5.1.0`.
- Run `sodev global:stop-all` to automatically run the latest migration to copy the new global docker-compose.yml (should see `★ Updating global docker-compose.yml...`) You should have a backup docker-compose.yml in `~/.config/squareone/global` and the new version should contain a healthcheck item the `mysql` section:
```
    healthcheck:
      test: mysqladmin ping -h mysql -u root --password=$$MYSQL_PASSWORD
      timeout: 3s
      retries: 20
```
- Remove the node version being used by SquareOne to mimic a fresh install: `nvm uninstall v14.15.1`.
- Run `sodev create` and the process should properly build the frontend and create the new databases for the project.
